### PR TITLE
Add check for existence of ECR images used as test dependencies in E2E EVM compatibility tests

### PR DESCRIPTION
**Pull request by lukaszcl**
The code changes include adding a new step in the workflow file `evm-version-compatibility-tests.yml` to check if the images used as test dependencies exist in the ECR. This step is triggered when there is a change in the test dependency or when the workflow is manually dispatched. The step checks if the latest Ethereum client image exists in the ECR and fails the workflow if it doesn't exist.

Here's [example of a failed workflow](https://github.com/smartcontractkit/chainlink/actions/runs/9255230733/job/25458610274) that resulted in triggering `update-internal-mirrors.yaml` workflow in CTF because there was no `ethereum/client-go:v1.14.3` image in our ECR.
image
 

- [x] Requires SECHD-14333 to be finished
- [x] Requires https://github.com/smartcontractkit/chainlink-testing-framework/pull/968

### CODE REVIEW

**Review by gheorghestrimtu — DISMISSED** [review]

File: .github/workflows/evm-version-compatibility-tests.yml [hunk 1]
```diff
@@ -55,11 +54,75 @@ jobs:
...
+        with:
+          repository: smartcontractkit/chainlink-testing-framework
+          path: chainlink-testing-framework
+      - name: Set up Go
+        uses: actions/setup-go@v3
+      - name: Run get-latest-ethereum-client-version ethereum/client-go:latest_stable command
+        id: get-latest-ethereum-client-version
+        run: |
+          cd chainlink-testing-framework/
+          ls utils/cmd/
```
Comment (Tofel) [inline .github/workflows/evm-version-compatibility-tests.yml]: remove?

File: .github/workflows/evm-version-compatibility-tests.yml [hunk 2]
```diff
@@ -55,11 +54,75 @@ jobs:
...
+          version=$(go run utils/cmd/main.go get-latest-ethereum-client-version ethereum/client-go:latest_stable)
+          echo "version=${version}" >> $GITHUB_OUTPUT
```
Comment (Tofel) [inline .github/workflows/evm-version-compatibility-tests.yml]: please print the version here as well, might be useful at some point

File: .github/workflows/evm-version-compatibility-tests.yml [hunk 3]
```diff
@@ -55,11 +54,75 @@ jobs:
...
+        with:
+          repository: ethereum/client-go
+          tag: ${{ steps.get-latest-ethereum-client-version.outputs.version }}
+          AWS_REGION: ${{ secrets.QA_AWS_REGION }}
+          AWS_ROLE_TO_ASSUME: ${{ secrets.QA_AWS_ROLE_TO_ASSUME }}
+      - name: Latest Ethereum client image does not exist in ECR
+        if: steps.check-latest-ethereum-client-image-exists.outputs.exists != 'true'
+        run: |
+          echo "ethereum/client-go:${{ steps.get-latest-ethereum-client-version.outputs.version }} does not exist in our ECR. The image download was scheduled, and you can rerun this workflow in few minutes."
+          exit 1
```
Comment (Tofel) [inline .github/workflows/evm-version-compatibility-tests.yml]: this is not a good idea, why can't we wait for mirror update job to finish?

File: .github/workflows/evm-version-compatibility-tests.yml [hunk 4]
```diff
@@ -55,11 +54,75 @@ jobs:
...
+        uses: smartcontractkit/.github/actions/setup-github-token@9e7cc0779934cae4a9028b8588c9adb64d8ce68c # setup-github-token@0.1.2
+        with:
+          aws-role-arn: ${{ secrets.AWS_OIDC_CHAINLINK_CI_INVOKE_CTF_WORKFLOWS_TOKEN_ISSUER_ROLE_ARN }}
+          aws-lambda-url: ${{ secrets.GATI_QA_LAMBDA_URL }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Trigger update-internal-mirrors workflow in CTF
+        run: gh workflow run --repo smartcontractkit/chainlink-testing-framework update-internal-mirrors.yaml
+        env:
+          GITHUB_TOKEN: ${{ steps.setup-github-token.outputs.access-token }}
+
```
Comment (Tofel) [inline .github/workflows/evm-version-compatibility-tests.yml:76]: please wait here for that job to finish (can be in a loop with sleep if there's no other way) and fail the workflow only if the update job fails

**Review by Tofel — CHANGES_REQUESTED** [review]
check the comments ;-)

File: .github/workflows/evm-version-compatibility-tests.yml (hunk 1 above)
Comment (lukaszcl) [reply .github/workflows/evm-version-compatibility-tests.yml]: removed

**Comment by lukaszcl** [comment]
@Tofel @gheorghestrimtu, please review again. I have created a GitHub Action to check for images and update the internal mirrors if necessary ([PR #968](https://github.com/smartcontractkit/chainlink-testing-framework/pull/968)). Now, if an update is needed, the workflows in core repo do not fail but instead wait until the images are updated

File: .github/workflows/evm-version-compatibility-tests.yml (hunk 2 above)
Comment (lukaszcl) [reply .github/workflows/evm-version-compatibility-tests.yml]: Fixed

File: .github/workflows/evm-version-compatibility-tests.yml (hunk 4 above)
Comment (lukaszcl) [reply .github/workflows/evm-version-compatibility-tests.yml:76]: Fixed

File: .github/workflows/evm-version-compatibility-tests.yml (hunk 3 above)
Comment (lukaszcl) [reply .github/workflows/evm-version-compatibility-tests.yml]: Fixed

**Review by gheorghestrimtu — DISMISSED** [review]

**Review by gheorghestrimtu — DISMISSED** [review]

File: .github/workflows/client-compatibility-tests.yml [hunk 5]
```diff
@@ -15,6 +15,23 @@ env:
...
+    name: Check images used as test dependencies exist in ECR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update internal ECR if the latest Ethereum client image does not exist
+        uses: smartcontractkit/chainlink-testing-framework/.github/actions/update-internal-mirrors@7eb04a030823b316d8dd5bb555f1e49593a503fc
+        with:
+          aws_region: ${{ secrets.QA_AWS_REGION }}
+          role_to_assume: ${{ secrets.QA_AWS_ROLE_TO_ASSUME }}
+          aws_account_number: ${{ secrets.QA_AWS_ACCOUNT_NUMBER }}
+          image_name: 'ethereum/client-go'
```
Comment (Tofel) [inline .github/workflows/client-compatibility-tests.yml]: this is too narrow, what about other clients?

**Review by Tofel — CHANGES_REQUESTED** [review]
we need to trigger that job for all images or all evm implementations, not only for go-ethereum

File: .github/workflows/client-compatibility-tests.yml (hunk 5 above)
Comment (lukaszcl) [reply .github/workflows/client-compatibility-tests.yml]: @Tofel what other clients you would like to check? Could you list them?

Comment (lukaszcl) [reply .github/workflows/client-compatibility-tests.yml]: @Tofel is this all we need?

```
          - name: ethereum/client-go
            expression: '^(alltools-v|v)[0-9]\.[0-9]+\.[0-9]+$'
          - name: hyperledger/besu
            expression: '^[0-9]+\.[0-9]+(\.[0-9]+)?$'
            page_size: 300
          - name: thorax/erigon
            expression: '^v[0-9]+\.[0-9]+\.[0-9]+$'
          - name: nethermind/nethermind
            expression: '^[0-9]+\.[0-9]+\.[0-9]+$'
          - name: tofelb/ethereum-genesis-generator
            expression: '^[0-9]+\.[0-9]+\.[0-9]+(\-slots\-per\-epoch)?'
```

Comment (lukaszcl) [reply .github/workflows/client-compatibility-tests.yml]: @Tofel I added it in https://github.com/smartcontractkit/chainlink/pull/13271/commits/bc9410c68e891a10e5323576653a53538b5a286d please check

**Review by gheorghestrimtu — DISMISSED** [review]

**Review by Tofel — APPROVED** [review]
👍

**Review by gheorghestrimtu — APPROVED** [review]

**Pull request merged**